### ISSUE TITLE
Threads running on evaluation?

### ISSUE THREAD

**Issue by vivekam101** [huggingface/transformers#1731]
Hi Transformers :)

Thanks for the wonderful repo. One quick question. I was running examples/test_examples.py and came to see logs printing multiple times while running in cpu. Is there threads spawned ? 
snippet from examples/test_examples.py
**
        testargs = ["run_squad.py",
                    "--train_file=./tests_samples/SQUAD/dev-v2.0-small.json",
                    "--predict_file=./tests_samples/SQUAD/dev-v2.0-small.json",
                    "--model_name=bert-base-uncased",
                    "--output_dir=./tests_samples/temp_dir",
                    "--max_steps=10",
                    "--warmup_steps=2",
                    "--do_train",
                    "--do_eval",
                    "--version_2_with_negative",
                    "--learning_rate=2e-4",
                    "--per_gpu_train_batch_size=2",
                    "--per_gpu_eval_batch_size=1",
                    "--overwrite_output_dir",
                    "--seed=42"]
***

Result where i see multiple logs prints
-----
11/05/2019 07:11:33 - INFO - utils_squad -   *** Example ***
*** Example ***
*** Example ***
11/05/2019 07:11:33 - INFO - utils_squad -   unique_id: 1000000013
unique_id: 1000000013
unique_id: 1000000013
11/05/2019 07:11:33 - INFO - utils_squad -   example_index: 13
example_index: 13
example_index: 13
11/05/2019 07:11:33 - INFO - utils_squad -   doc_span_index: 0
doc_span_index: 0
doc_span_index: 0

**Comment by vivekam101** [comment]
*attaching complete log
[test_log.txt](https://github.com/huggingface/transformers/files/3806959/test_log.txt)

**Comment by vivekam101** [comment]
Sorry. guys. logger has been called multiple times hence the issue closing the issue.
